python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -160,6 +160,9 @@ def __setattr__(self, name: str, value: Any):
     def _is_field_internal(self, name: str) -> bool:
         return name.endswith(INTERNAL_MARKER)
 
+    def _is_field_private(self, name: str) -> bool:
+        return name.startswith("_")
```

Comment:
Specifically I don't think we should do this and instead we should just change the base class definition to tell Pydantic to allow this.

Reply:
That seems reasonable.

Right now, the model doesn't support assignment because it is `frozen = True`. If we disable that, we'll still hit the `ValueError` for trying to initialize unspecified fields, which I think will still require some custom `__setattr__` handling to get around

Reply:
Unless we use `PrivateAttr` - that seems to work ok, based on [this discussion](https://github.com/pydantic/pydantic/issues/655#issuecomment-1289635676):

```python
class PrometheusResource(ConfigurableResource):
  gateway: str = Field(
      description=(
          "The url for your push gateway. Either of the"
          " form 'http://pushgateway.local', or 'pushgateway.local'."
          " Scheme defaults to 'http' if none is provided"
      )
  )
  timeout: int = Field(
      default=30,
      description="is how long delete will attempt to connect before giving up. Defaults to 30s.",
  )
  _registry: prometheus_client.CollectorRegistry = PrivateAttr(default=None)
  
  def __post_init__(self) -> None:
      self._registry = prometheus_client.CollectorRegistry()
```